src/main/java/com/j256/ormlite/dao/BaseDaoImpl.java

```diff
@@ -372,12 +373,23 @@ public Integer call() throws SQLException {
 	}
 
 	@Override
-	public synchronized T createIfNotExists(T data) throws SQLException {
-		if (data == null) {
+	public synchronized T createIfNotExists(final T data) throws SQLException {
```

Comment:
This creates an unnecessary object on every call.  I don't see a lot of benefit to changing this method to call over to the `Supplier` method.